encoding/codecv2/codecv2.go

```diff
@@ -193,6 +193,20 @@ func ConstructBlobPayload(chunks []*encoding.Chunk, useMockTxData bool) (*kzg484
 		}
 	}
 
+	if conditionalEncode {
+		encoded := len(blobBytes) < len(batchBytes)
+		if encoded {
```

Comment:
We should only compress and check compatibility if the caller requested `encoded = true`.

Reply:
Yes. this is an outdated implementation. will refactor to let rollup-relayer decide if it needs to be encoded.